conan/cache/cache.py

```diff
@@ -18,17 +18,16 @@ class DataCache:
 
     def __init__(self, base_folder, db_filename):
         self._base_folder = os.path.realpath(base_folder)
-        self.db = CacheDatabase(filename=db_filename)
-        self.db.initialize(if_not_exists=True)
```

Comment:
A bit "spagetti" code, CacheDatabase will fail to work if not initialize() is called, then call it directly in the constructor. Encapsulate until it is necessary to extract.